src/wasm/wasm.cpp

```diff
@@ -931,15 +931,29 @@ void RefCast::finalize() {
 
 Type RefCast::getCastType() { return doGetCastType(this); }
 
-void BrOnCast::finalize() {
-  if (ref->type == Type::unreachable || rtt->type == Type::unreachable) {
+void BrOn::finalize() {
+  if (ref->type == Type::unreachable ||
+      (rtt && rtt->type == Type::unreachable)) {
```

Comment:
Is it ever ok for `rtt` to be unreachable in practice? If not, perhaps this should be an assert.

Reply:
This is a little tricky. The text and binary readers will not produce such a thing. But creating one as IR directly could. I think it's better to let the validator catch the error and print it nicely, rather than assert here, but I could go either way - thoughts?

Reply:
Ok, that's fine with me. I guess the worst case is that someone sees a crash when they try to get the rtt type of an unreachable and that shouldn't leave much question about what the problem is.